### PR TITLE
contrib/dbus-test-runner: new package (19.04.0)

### PR DESCRIPTION
**Pull request by anixedn**
Required by https://github.com/chimera-linux/cports/pull/1385
I disabled checks for now, because i can't really figure out, why does it fail
Here is the log with results https://pastebin.com/ZWk3pGCQ

### CODE REVIEW

**Comment by q66** [comment]
i see only one FAIL and that's test-output, which fails because of gnu assumptions in head (normally it's not allowed to take a negative number): https://github.com/ubports/dbus-test-runner/blob/86d63d119566974bd841cacd3202599b3b1a845d/tests/Makefile.am#L162

the purpose is to cut off the last line, which you can do with `sed '$d'`

File: contrib/dbus-test-runner/template.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+]
+makedepends = [
+    "dbus",
+    "dbus-glib-devel",
+    "gettext-devel",
+    "glib-devel",
+    "glibmm-devel",
+]
+checkdepends = ["bash", "python-dbusmock"]
+pkgdesc = (
```
Comment (q66) [inline contrib/dbus-test-runner/template.py:23]: this is too long, something like `Run executables under a new DBus session` is enough

File: contrib/dbus-test-runner/template.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+hostmakedepends = [
+    "autoconf",
+    "automake",
+    "gmake",
+    "intltool",
+    "libtool",
+    "pkgconf",
+]
+makedepends = [
+    "dbus",
```
Comment (q66) [inline contrib/dbus-test-runner/template.py:16]: dbus should go in `depends` and `checkdepends`

File: contrib/dbus-test-runner/template.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    "dbus-glib-devel",
+    "gettext-devel",
+    "glib-devel",
+    "glibmm-devel",
```
Comment (q66) [inline contrib/dbus-test-runner/template.py:20]: i don't think glibmm is necessary at all

File: contrib/dbus-test-runner/template.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "automake",
+    "gmake",
+    "intltool",
+    "libtool",
+    "pkgconf",
+]
+makedepends = [
+    "dbus",
+    "dbus-glib-devel",
+    "gettext-devel",
```
Comment (q66) [inline contrib/dbus-test-runner/template.py:18]: gettext is not necessary either

File: contrib/dbus-test-runner/template.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    "glib-devel",
+    "glibmm-devel",
+]
+checkdepends = ["bash", "python-dbusmock"]
+pkgdesc = (
+    "Little utility to test a couple of executables under a new DBus session"
+)
+maintainer = "avgwst <avgwst@tutanota.de>"
+license = "GPL-3.0-or-later"
```
Comment (q66) [inline contrib/dbus-test-runner/template.py:27]: this is GPL-3.0-only

File: contrib/dbus-test-runner/template.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    "autoconf",
+    "automake",
+    "gmake",
+    "intltool",
+    "libtool",
+    "pkgconf",
+]
+makedepends = [
+    "dbus",
+    "dbus-glib-devel",
```
Comment (q66) [inline contrib/dbus-test-runner/template.py:17]: dbus-glib is seemingly not even used, you just need to patch it out of configure.ac

File: contrib/dbus-test-runner/patches/fix-test-output.patch [hunk 7]
```diff
@@ -0,0 +1,27 @@
+diff --git a/tests/Makefile.am b/tests/Makefile.am
+index 4f4aad9..5e2d0b7 100644
+--- a/tests/Makefile.am
++++ b/tests/Makefile.am
+@@ -159,17 +159,17 @@ test-output: Makefile.am
+ 	@echo "#!/bin/sh -e" > $@
+ 	@echo "$(DBUS_RUNNER) --task cat --parameter \"$(top_srcdir)/src/dbus-test-runner.c\" --task-name \"cat1\" --task cat --parameter \"$(top_srcdir)/src/dbus-test-runner.c\" --task-name \"cat2\" > testcat.output.txt" >> $@
```
Comment (q66) [inline contrib/dbus-test-runner/patches/fix-test-output.patch:7]: the test-output test is flaking here, often cat2 does not manage to print out anything (i guess it races)

maybe best to just disable test-output from the suite

File: contrib/dbus-test-runner/template.py (hunk 4 above)
Comment (anixedn) [reply contrib/dbus-test-runner/template.py:18]: should i also patch for not using gettext? configure requires gettext-devel

File: contrib/dbus-test-runner/template.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    "intltool",
+    "libtool",
+    "pkgconf",
+]
+makedepends = [
+    "gettext-devel",
+    "glib-devel",
+]
+depends = ["dbus"]
+checkdepends = ["dbus", "bash", "python-dbusmock"]
```
Comment (q66) [inline contrib/dbus-test-runner/template.py:19]: sort

File: contrib/dbus-test-runner/template.py (hunk 4 above)
Comment (q66) [reply contrib/dbus-test-runner/template.py:15]: i guess leave it, but move it to hostmakedepends

**Pull request merged**